### PR TITLE
nixos/hbase: add settings option for hbase-site.xml

### PR DESCRIPTION
**Pull request by VergeDX**
###### Motivation for this change
HBase master info port is the Web UI used.
If not specify in config, it will use the random one.

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/databases/hbase.nix [hunk 1]
```diff
@@ -85,6 +89,14 @@ in {
         '';
       };
 
+      masterInfoPort = mkOption {
+        type = types.int;
```
Comment (aanderse) [inline nixos/modules/services/databases/hbase.nix:93]: There is a `types.port`.

**Review by aanderse — COMMENTED** [review]
This works... but after reading up on this file wouldn't it be much more fun if instead we added a `settings` option and users could configure the entire file? Like so:
```nix
services.hbase.settings = {
  "hbase.master.info.port" = 60010;
  "hbase.cluster.distributed" = true;
  # any property here! etc...
};
```

Does that sound appealing to you?

**Review by aanderse — APPROVED** [review]
This looks great! I'll merge now if you're happy with this... but I do wonder if there is value is specifying a stricter type, like `attrsOf (oneOf [ str int bool ])`.

Let me know. Thanks!

**Comment by VergeDX** [comment]
You're right, the example config seems only contains two types of `int` and `bool`.
We also needs `str` types to put `logDir` & `dataDir`, changes performed.

**Comment by aanderse** [comment]
Thanks @VergeDX!

**Pull request merged**